File: Algorithms/src/1855. Maximum Distance Between a Pair of Values/Code/solution.go
func maxDistance(nums1 []int, nums2 []int) int {
    i, j := 0, 0
    ans := 0

    for i < len(nums1) && j < len(nums2) {

        // Ensure i <= j
        if i > j {
            j++
            continue
        }

        // Valid pair
        if nums1[i] <= nums2[j] {
            if j-i > ans {
                ans = j - i
            }
            j++ // Try for larger distance
        } else {
            // Invalid pair
            i++
        }
    }

    return ans
}

File: Algorithms/src/1855. Maximum Distance Between a Pair of Values/Code/solution.py
class Solution:
    def maxDistance(self, nums1: List[int], nums2: List[int]) -> int:
        i = 0
        j = 0
        ans = 0

        while i < len(nums1) and j < len(nums2):

            # Ensure i <= j
            if i > j:
                j += 1
                continue

            # Valid pair
            if nums1[i] <= nums2[j]:
                ans = max(ans, j - i)
                j += 1  # Try for larger distance
            else:
                # Invalid pair
                i += 1

        return ans